### PR TITLE
Highload merge

### PR DESCRIPTION
**Pull request by litinsky**
1) Created agent, service and channel gateway entities;
2) Implemented RabbitMQ agent, service and channel gateways;
3) Implemented `service` mode to run transport wrapper for service;
4) Implemented logging, moved service response time logging configuration to `log_config.yml` along with general logger configuration.

`agent` and `channel` running modes are not implemented yet.

### CODE REVIEW

File: core/transport/messages.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    @classmethod
+    def from_json(cls, message_json):
+        message_json.pop('msg_type')
+        return cls(**message_json)
+
+    def to_json(self) -> dict:
+        return self.__dict__
+
+
+TMessageBase = TypeVar('TMessageBase', bound=MessageBase)
```
Comment (my-master) [inline core/transport/messages.py]: it's better to put all types to the bottom of the file

File: core/transport/base.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        self._on_channel_callback = callback
+
+    async def send_to_service(self, service: str, dialog: Dict) -> None:
+        raise NotImplementedError
+
+    async def send_to_channel(self, channel_id: str, user_id: str, response: str) -> None:
+        raise NotImplementedError
+
+
+TAgentGateway = TypeVar('TAgentGateway', bound=AgentGatewayBase)
```
Comment (my-master) [inline core/transport/base.py]: it's better to put all types to the bottom of the file

File: core/transport/base.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+
+class ServiceGatewayConnectorBase:
+    _service_config: dict
```
Comment (my-master) [inline core/transport/base.py]: types should be literals like `Dict`

File: core/run.py [hunk 4]
```diff
@@ -225,5 +238,42 @@ def main():
...
+
+def main():
+    if MODE == 'default':
+        run_default()
+    elif MODE == 'agent':
+        run_agent()
+    elif MODE == 'service':
+        run_service()
+    elif MODE == 'channel':
+        run_channel()
```
Comment (my-master) [inline core/run.py]: This section doesn't look good. It should not separate 'default' and other things.

File: core/transport/messages.py (hunk 1 above)
Comment (litinsky) [reply core/transport/messages.py]: But it is used in type annotation further in this file.

File: core/connectors.py [hunk 5]
```diff
@@ -108,3 +110,39 @@ def __init__(self, service_name: str):
...
+        await self._to_service_callback(dialog=payload, service_name=self._service_name)
+
+
+class ServiceGatewayHTTPConnector(ServiceGatewayConnectorBase):
+    _session: aiohttp.ClientSession
+    _url: str
+    _service_name: str
+
+    def __init__(self, service_config: dict, formatter: Callable) -> None:
+        super(ServiceGatewayHTTPConnector, self).__init__(service_config, formatter)
```
[the remark this answers is not in the document]
Comment (litinsky) [reply core/connectors.py]: will fix

File: core/config_parser.py [hunk 6]
```diff
@@ -106,45 +133,73 @@ def add_bot_to_name(name):
...
+
+    for config in chain(SKILLS, ANNOTATORS_1, ANNOTATORS_2, ANNOTATORS_3,
+                        SKILL_SELECTORS, RESPONSE_SELECTORS, POSTPROCESSORS):
+        config_name = config['name']
+
+        if config_name == service_name:
+            matching_config = config
+            break
+
+    if not matching_config:
```
[the remark this answers is not in the document]
Comment (litinsky) [reply core/config_parser.py]: done

File: core/connectors.py [hunk 7]
```diff
@@ -108,3 +110,39 @@ def __init__(self, service_name: str):
...
+        self._session = aiohttp.ClientSession()
+        self._service_name = service_config['name']
+
+        default_infer_url = 'http://127.0.0.1:5000/model'
```
[the remark this answers is not in the document]
Comment (litinsky) [reply core/connectors.py]: fixed

File: core/connectors.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (litinsky) [reply core/connectors.py]: fixed

File: core/transport/base.py (hunk 2 above)
Comment (litinsky) [reply core/transport/base.py]: In this file it is ok. Moved.

File: core/transport/base.py (hunk 3 above)
Comment (litinsky) [reply core/transport/base.py]: fixed

File: core/run.py (hunk 4 above)
Comment (litinsky) [reply core/run.py]: decided to leave in this PR

**Pull request merged**